rust/feature-flags/src/request_handler.rs

```diff
@@ -189,29 +216,24 @@ fn decode_request(headers: &HeaderMap, body: Bytes) -> Result<FlagRequest, FlagE
 ///  and that flag will be omitted from the result (we will still attempt to evaluate other flags)
 // TODO: it could be a cool idea to store the errors as a tuple instead of top-level, so that users can see
 // which flags failed to evaluate
-pub async fn evaluate_feature_flags(
-    team_id: i32,
-    distinct_id: String,
-    feature_flags_from_cache_or_pg: FeatureFlagList,
-    database_client: Arc<dyn Client + Send + Sync>,
-    person_property_overrides: Option<HashMap<String, Value>>,
-    group_property_overrides: Option<HashMap<String, HashMap<String, Value>>>,
-    groups: Option<HashMap<String, Value>>,
-) -> FlagsResponse {
-    let group_type_mapping_cache = GroupTypeMappingCache::new(team_id, database_client.clone());
+pub async fn evaluate_feature_flags(context: FeatureFlagEvaluationContext) -> FlagsResponse {
+    let group_type_mapping_cache =
+        GroupTypeMappingCache::new(context.team_id, context.postgres_reader.clone());
     let mut feature_flag_matcher = FeatureFlagMatcher::new(
-        distinct_id.clone(),
-        team_id,
-        database_client,
+        context.distinct_id,
+        context.team_id,
+        context.postgres_reader,
+        context.postgres_writer,
         Some(group_type_mapping_cache),
-        None,
-        groups,
+        None, // TODO maybe remove this from the matcher struct, since it's used internally but not passed around
```

Comment:
this is for the `PropertyCache` which I use to track properties fetched from the DB during the lifetime of the request.  It's never instantiated initially, so it feels non-ergonomic to expose it on the API, hence the TODO.